drivers/uart_half_duplex/include/uart_half_duplex.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_uart_half_duplex half-duplex UART Driver
+ * @ingroup     drivers_actuators
+ *
+ * This module contains drivers for UART half-duplex communication bus.
+ * It needs to manage the communication direction by enabling or disabling TX.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for half-duplex UART driver
+ *
+ * @author      Loïc Dauphin <loic.dauphin@inria.fr>
+ */
+
+#ifndef UART_HALF_DUPLEX_H
+#define UART_HALF_DUPLEX_H
+
+#include <stdlib.h>
+
+#include "periph/uart.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef UART_HALF_DUPLEX_DEFAULT_TIMEOUT_US
+#define UART_HALF_DUPLEX_DEFAULT_TIMEOUT_US (20000) /**< Default recv timeout (in microseconds) */
+#endif
+
+/**
+ * @brief half-duplex UART direction management method type
+ */
+typedef int uart_half_duplex_dir_t;
+
+#define UART_HALF_DUPLEX_DIR_NONE           (0)  /**< Don't manage direction */
+
+#define UART_HALF_DUPLEX_DIR_PIN_SET(pin)   (((pin + 1) << 1))     /**< pin set enables TX */
+#define UART_HALF_DUPLEX_DIR_PIN_CLEAR(pin) (((pin + 1) << 1) | 1) /**< pin clear enables TX */
+
+/**
+ * @brief Descriptor struct for half-duplex UART
+ */
+typedef struct {
+    uart_t uart;                 /**< the half-duplex UART bus used */
```

Comment:
attributes seems duplicated from the params struct, can you replace them a params member ?